src/qibolab/platforms/abstract.py

```diff
@@ -15,6 +18,68 @@
 )
 
 
+@dataclass
+class Qubit:
+    name: str
+    readout_frequency: int = 0
+    drive_frequency: int = 0
+    sweetspot: float = 0
+    peak_voltage: float = 0
+    pi_pulse_amplitude: float = 0
+    T1: int = 0
+    T2: int = 0
+    state0_voltage: int = 0
+    state1_voltage: int = 0
+    mean_gnd_states: complex = 0 + 0.0j
+    mean_exc_states: complex = 0 + 0.0j
+    resonator_polycoef_flux: List[float] = field(default_factory=list)
+
+
+@dataclass
+class ExecutionResults:
+    """Data strucutre to store output from `execute_pulse_sequence`."""
+
+    i: np.ndarray
+    q: np.ndarray
+    sample: np.ndarray
```

Comment:
If I understand correctly this will hold the single shots. How about we call it
```suggestion
    samples: np.ndarray
```
or
```suggestion
    shots: np.ndarray
```
since in most (if not all) cases it will have more than one shot.